sensors/simsensor_common/simsensor_reader.c

```diff
@@ -0,0 +1,286 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Simsensor reader reads and parses data form special scenario file
+ * than it pushes it to event_queue
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Piotr Nieciecki
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <limits.h>
+#include <sys/types.h>
+#include <string.h>
+#include <sys/time.h>
+
+#include "simsensor_reader.h"
+
+#define READER_END_SCENARIO_INDICATOR NO_SENSOR
+#define READER_DATA_SEPARATOR         ','
+
+#define READER_TIMESTAMP_NOT_SET (time_t) LLONG_MAX
+
+
+static int reader_getFieldLLong(const char *actField, char **nextField, long long *res)
+{
+	char *endptr;
+	long long value = strtoll(actField, &endptr, 10);
+	if (!(*endptr == READER_DATA_SEPARATOR || *endptr == '\0' || *endptr == '\n') || actField == endptr) {
+		printf("Wrong file\n");
+		return -1;
+	}
+
+	if (*endptr == READER_DATA_SEPARATOR && nextField != NULL) {
+		*nextField = endptr + 1;
+	}
+
+	*res = value;
+
+	return 0;
+}
+
+
+static int reader_accelDataParse(char *startOfDataSection, time_t timestamp, sensor_event_t *result)
+{
+	char *actField, *nextField;
+	long long tmp;
+
+	actField = startOfDataSection;
+
+	result->type = SENSOR_TYPE_ACCEL;
+	result->timestamp = timestamp;
+
+	if (reader_getFieldLLong(actField, &nextField, &tmp) < 0) {
+		return -1;
+	}
+	result->accels.accelX = tmp;
+	actField = nextField;
+
+	if (reader_getFieldLLong(actField, &nextField, &tmp) < 0) {
+		return -1;
+	}
+	result->accels.accelY = tmp;
+	actField = nextField;
+
+	if (reader_getFieldLLong(actField, NULL, &tmp) < 0) {
+		return -1;
+	}
+	result->accels.accelZ = tmp;
+
+	return 0;
+}
+
+
+static int reader_baroDataParse(char *startOfDataSection, time_t timestamp, sensor_event_t *result)
+{
+	char *actField, *nextField;
+	long long tmp;
+
+	actField = startOfDataSection;
+
+	result->type = SENSOR_TYPE_BARO;
+	result->timestamp = timestamp;
+
+	if (reader_getFieldLLong(actField, &nextField, &tmp) < 0) {
+		return -1;
+	}
+	result->baro.pressure = tmp;
```

Comment:
Why do we need `tmp`? On error we leave `result` struct destroyed anyway (e.g. first field is ok, next one is not). Applies to other places too

Reply:
We use `tmp`, because `reader_getFieldLLong` takes pointer to `long long` as an argument and data what we want to read is not always a `long long`. For example there `baro.pressure` is `unit32_t`

Reply:
Perhaps this field should be read a `long` instead of `long long` then, but ack